packages/fxa-auth-server/lib/routes/mfa.ts

```diff
@@ -0,0 +1,339 @@
+        success = true;
+      }
+    } catch (error) {
+      // TODO: Handle specific OTP verification errors
+
+      throw AppError.backendServiceFailure(
+        'OtpService',
+        'verifyCode',
+        { uid },
+        error
+      );
+    }
+
+    if (success) {
+      // TODO: Create glean metric
+      // await this.glean.otp.verifyOtpSuccess(request);
+
+      const account = await this.db.account(uid);
+
+      this.statsd.increment('account.otp.verifyOtp.success');
+
+      // TODO: At some point we might want to create a mapping between actions and scopes.
+      // this would allow us to let multiple actions to all be sanctioned by a single jwt.
+      const scope = action;
+
+      // Issue jwt
+      const now = Math.floor(Date.now() / 1000);
+      const claims = {
+        sub: account.uid,
+        scope: [`mfa:${scope}`],
+        iat: now,
+        jti: uuid.v4(),
+      };
+
+      const key = this.config.mfa.jwt.secretKey;
+      const opts = {
+        algorithm: 'HS256' as jwt.Algorithm,
+        expiresIn: this.config.mfa.jwt.expiresInSec,
+        audience: this.config.mfa.jwt.audience,
+        issuer: this.config.mfa.jwt.issuer,
+      } as jwt.SignOptions;
+
+      // Create access token
+      const accessToken: string = jwt.sign(claims, key, opts);
+
+      await recordSecurityEvent('account.mfa_verify_otp_code_success', {
+        db: this.db,
+        request,
+      });
+
+      return { accessToken };
+    }
+
+    await recordSecurityEvent('account.mfa_verify_otp_code_failed', {
+      db: this.db,
+      request,
+    });
+
+    throw AppError.invalidOrExpiredOtpCode();
+  }
+}
+
+export const mfaRoutes = (
+  customs: Customs,
+  db: any,
+  // TODO: Add glean metrics
+  // glean: GleanMetricsType,
+  log: any,
+  mailer: any,
+  statsd: any,
+  config: ConfigType
+) => {
+  const featureEnabledCheck = () => {
+    if (!config.mfa.enabled) {
+      throw AppError.featureNotEnabled();
+    }
+    return true;
+  };
+
+  const otpUtils = Container.get(OtpUtils);
+
+  const otpHandler = new MfaHandler(
+    config,
+    db,
+    otpUtils,
+    customs,
+    mailer,
+    statsd
+  );
+
+  const routes = [
+    {
+      method: 'POST',
+      path: '/mfa/otp/request',
+      options: {
+        pre: [{ method: featureEnabledCheck }],
+        auth: {
+          strategy: 'sessionToken',
+        },
+        validate: {
+          payload: isA.object({
+            action: isA.string().required(),
+          }),
+        },
+      },
+      handler: function (request: AuthRequest) {
+        log.begin('mfa.otp.request', request);
+        return otpHandler.requestOtpCode(request);
+      },
+    },
+    {
+      method: 'POST',
+      path: '/mfa/otp/verify',
+      options: {
+        auth: {
+          strategy: 'sessionToken',
+        },
+        validate: {
+          payload: isA.object({
+            code: isA.string().required(),
+            action: isA.string().required(),
```

Comment:
Should we have allowed values for the `action` type to start off?

Reply:
Yeah, that's a good call. Otherwise it might be easy to overlook in the future. Let me change this.